src/Http/Http/src/Internal/ResponseCookies.cs

```diff
@@ -154,17 +154,25 @@ public void Delete(string key, CookieOptions options)
             }
 
             var encodedKeyPlusEquals = (_enableCookieNameEncoding ? Uri.EscapeDataString(key) : key) + "=";
-            bool domainHasValue = !string.IsNullOrEmpty(options.Domain);
-            bool pathHasValue = !string.IsNullOrEmpty(options.Path);
+            bool domainAndPathHasValue = !string.IsNullOrEmpty(options.Domain) && !string.IsNullOrEmpty(options.Path);
+            bool domainOnlyHasValue = !string.IsNullOrEmpty(options.Domain) && string.IsNullOrEmpty(options.Path);
+            bool pathOnlyHasValue = !string.IsNullOrEmpty(options.Path) && string.IsNullOrEmpty(options.Domain);
 
             Func<string, string, CookieOptions, bool> rejectPredicate;
-            if (domainHasValue)
+            if (domainAndPathHasValue)
```

Comment:
Did the variables above need to change? Could this new condition have been like this?
```suggestion
            if (domainHasValue && pathHasValue)
```
The else's already imply 'only'.

Reply:
Yep you're right :slightly_smiling_face:. We can do the first condition with **` if (domainHasValue && pathHasValue)`**.